wp/lib/bap_wp/src/environment.mli

```diff
@@ -114,6 +114,10 @@ val wp_rec_call :
     typically a constant. *)
 val add_var : t -> Bap.Std.Var.t -> Constr.z3_expr -> t
 
+(** Add a function predicate to the environment representing a function call to
+    a target with a given tid. *)
+val add_fun_pred : t -> Bap.Std.Tid.t -> Constr.z3_expr -> t
```

Comment:
So the arguments are fixed by the shape of the function call? Maybe we should elaborate a bit in the comments here.

Reply:
I'm not sure I understand which arguments you are referring to. Are you referring to the `z3_expr` argument that takes in a predicate of the form `foo(in, ..., out, ...)`?

Reply:
I was actually referring to the `tid` input; generally, describing what all the non-trivial arguments are supposed to be is a good idea.

Reply:
So the `tid` is a function call site, and the `z3_expr` is the predicate that represents that function call?

Reply:
Updated the doc with more info about the `tid` and `z3_expr`.